include/crm/common/strings_internal.h

```diff
@@ -16,6 +16,13 @@
 
 /* internal generic string functions (from strings.c) */
 
+enum pcmk__str_flags {
+    pcmk__str_none          = 0,
+    pcmk__str_casei         = 1 << 0,
+    pcmk__str_null_matches  = 1 << 1,
+    pcmk__str_regex         = 1 << 2
```

Comment:
Something that was suggested in another PR was to use the UINTn_C() macros from stdint.h, when the mask using the enum will be a stdint.h type, e.g. UINT32_C(1) << 0

It's not really necessary in our current usage, but would future-proof against any unusual arches and follows how the stdint types are meant to be used.

On the other hand, it sort of forces a mask length when defining the enum, whereas it's conceivable that a mask in one place could be uint32_t and a mask in another place could be uint64_t. So I'm ambivalent about it. That's less of a concern for internal enums though since we can change them if needed.